data/json/terrain.json

```diff
@@ -7704,6 +8008,69 @@
     "flags": [ "TRANSPARENT", "FLAT" ],
     "bash": { "str_min": 30, "str_max": 210, "sound": "crash!", "sound_fail": "whump!", "ter_set": "t_null", "bash_below": true }
   },
+  {
+    "type": "terrain",
+    "id": "t_tar_flat_roof",
+    "name": "tar paper flat roof",
+    "looks_like": "t_flat_roof",
+    "symbol": ".",
+    "color": "gray",
+    "move_cost": 2,
+    "flags": [ "TRANSPARENT", "FLAMMABLE", "FLAT" ],
+    "bash": { "str_min": 30, "str_max": 210, "sound": "crash!", "sound_fail": "whump!", "ter_set": "t_null", "bash_below": true }
+  },
+  {
+    "type": "terrain",
+    "id": "t_shingle_flat_roof",
+    "name": "shingle flat roof",
+    "looks_like": "t_flat_roof",
+    "symbol": ".",
+    "color": "green",
+    "move_cost": 2,
+    "flags": [ "TRANSPARENT", "FLAMMABLE", "FLAT" ],
+    "bash": { "str_min": 30, "str_max": 210, "sound": "crash!", "sound_fail": "whump!", "ter_set": "t_null", "bash_below": true }
```

Comment:
Should the ter_set be t_null, or t_open_space for a broken roof?
Or should we create an option like t_rafters that you can fall through that only appears in houses under construction or underneath a broken roof?

Reply:
I will set all flat_roofs to open_air for now (falling through makes sense), but if we make floor/roof tiles into floor/rafters in a later PR, these can be switched to the t_rafters terrain.  Seems like it could be part of a larger change/idea.